src/main/scala/net/cakesolutions/CakeBuildInfoPlugin.scala

```diff
@@ -25,9 +33,54 @@ object CakeBuildInfoPlugin extends AutoPlugin {
         Try("git rev-parse --verify HEAD".!! dropRight 1) getOrElse "n/a"
       )
     ),
-    buildInfoPackage := s"${organization.value}.${(name in ThisBuild).value}.build",
+    buildInfoPackage :=
+      s"${organization.value}.${(name in ThisBuild).value}.build",
     buildInfoOptions += BuildInfoOption.BuildTime,
     buildInfoOptions += BuildInfoOption.ToJson
   )
 
 }
+
+/**
+  * Build keys that will be auto-imported when this plugin is enabled.
+  */
+object CakeBuildInfoKeys {
+  /**
+    * Map holding generic set of labelled values. These are use to label jar
+    * manifests and Docker containers.
+    */
+  val generalInfo: Map[String, String] = Map(
+    "buildDatetime" -> buildDatetime,
+    "buildHost" -> buildHost,
+    "buildUser" -> buildUser,
+    "gitBranch" -> gitBranch,
+    "gitCommitHash" -> gitCommitHash,
+    "gitRepository" -> gitRepository
+  )
+
+  /**
+    * Map holding Docker specific labelled values. These are used to label
+    * Docker containers.
+    */
+  val dockerInfo: Map[String, String] = Map(
+    "buildDockerVersion" -> buildDockerVersion
+  )
+
+  private def buildDatetime =
+    ZonedDateTime.now(Clock.systemUTC()).format(DateTimeFormatter.ISO_INSTANT)
+
+  private def buildDockerVersion = "docker --version".!!.trim
+
+  private def buildHost = "hostname".!!.trim
+
+  private def buildUser = "whoami".!!.trim
+
+  private def gitBranch = (
+    // Select the current branch (the one with asterisk next to it)
+    "git branch" #| "grep *"
+    ).!!.trim.drop(2) // Get rid of the asterisk to get the branch name only
+
+  private def gitCommitHash = "git rev-parse HEAD".!!.trim
```

Comment:
Looking at the `DynVerPlugin` source, I'm not so sure it is? That plugin appears to use a `git describe` (+ some extra params) to build its value for the SBT setting `dynver`. Aren't we after something slightly different here?